awx/main/scheduler/task_manager.py

```diff
@@ -426,82 +420,45 @@ def should_update_related_project(self, job, latest_project_update):
             return True
         return False
 
-    def generate_dependencies(self, task):
+    def generate_dependencies(self, pending_tasks):
         dependencies = []
-        if type(task) is Job:
-            # TODO: Can remove task.project None check after scan-job-default-playbook is removed
-            if task.project is not None and task.project.scm_update_on_launch is True:
-                latest_project_update = self.get_latest_project_update(task)
-                if self.should_update_related_project(task, latest_project_update):
-                    project_task = self.create_project_update(task)
-                    dependencies.append(project_task)
-                else:
-                    if latest_project_update.status in ['waiting', 'pending', 'running']:
-                        dependencies.append(latest_project_update)
-
-            # Inventory created 2 seconds behind job
-            try:
-                start_args = json.loads(decrypt_field(task, field_name="start_args"))
-            except ValueError:
-                start_args = dict()
-            for inventory_source in [invsrc for invsrc in self.all_inventory_sources if invsrc.inventory == task.inventory]:
-                if "inventory_sources_already_updated" in start_args and inventory_source.id in start_args['inventory_sources_already_updated']:
-                    continue
-                if not inventory_source.update_on_launch:
-                    continue
-                latest_inventory_update = self.get_latest_inventory_update(inventory_source)
-                if self.should_update_inventory_source(task, latest_inventory_update):
-                    inventory_task = self.create_inventory_update(task, inventory_source)
-                    dependencies.append(inventory_task)
-                else:
-                    if latest_inventory_update.status in ['waiting', 'pending', 'running']:
-                        dependencies.append(latest_inventory_update)
+        for task in pending_tasks:
+            if type(task) is Job and not task.dependent_jobs.exists():
+                # TODO: Can remove task.project None check after scan-job-default-playbook is removed
+                if task.project is not None and task.project.scm_update_on_launch is True:
+                    latest_project_update = self.get_latest_project_update(task)
```

Comment:
It's surprising to me that `get_latest_project_update` makes a DB query to get it, but that's just the way it is. It is an appealing optimization to check if there is a project update for that `task.project` inside of the running list `dependencies`, you might have mentioned that to me before, but it's probably a fairly rare circumstance relative to everything going on.